src/hdmf/build/map.py

```diff
@@ -415,6 +415,8 @@ def convert_dtype(cls, spec, value):
                  The value is returned as the function may convert the input value to comply
                  with the dtype specified in the schema.
         """
+        if isinstance(value, DataChunkIterator):
+            return value, None
```

Comment:
Here you would need to check for ``AbstractDataChunkIterator`` instead. However, this does not fix the issue, because in this case data with a dtype that is incompatible with the spec can be written. I'm working on a fix for this issue right now and add a PR in a bit once I've fixed it.